### PR TITLE
Issue56 dep tests

### PR DESCRIPTION
**Pull request by technetos**
Its up to us if we want to merge this in now or merge it in along with the refactor.  If we merge this in to dev now we will have a more concise history.  

The only reason its hasnt been merged yet is because of the missing tests for cloning but thats going to take longer than I would like.  

I plan to make another pass at implementing tests that involve actually cloning projects down when we have project structures to be cloned down.  

Implemented tests for the dependencies code.

### CODE REVIEW

File: hatch/src/deps/mod.rs [hunk 1]
```diff
@@ -1,3 +1,7 @@
+#[cfg(test)]
+mod tests;
+
+
```
Comment (DannyPeck) [inline hatch/src/deps/mod.rs]: Can we get rid of this extra new line?

File: hatch/src/deps/tests/clone_one_dep.rs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+use deps;
+use project::Dependency;
+
+#[test]
+fn clone_one_dep() {
+  let registry = Path::new("TestProject/hatch_modules");
+
```
Comment (DannyPeck) [inline hatch/src/deps/tests/clone_one_dep.rs]: Can we get rid of these new lines?

File: hatch/src/deps/tests/dependency.rs [hunk 3]
```diff
@@ -0,0 +1,23 @@
+use project::Dependency;
+
+#[test]
+fn dependency_name() {
+  let dep =
```
Comment (DannyPeck) [inline hatch/src/deps/tests/dependency.rs]: This is weird, maybe put the string on a different line and then pass it in as a variable to conserve space?

File: hatch/src/deps/tests/dependency.rs [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  assert_eq!(expected_name, actual_name); 
+}
+
+#[test]
+fn dependency_url() {
+  let dep =
+    Dependency::new("https://github.com/CachedNerds/TestProject.git".to_owned());
+
+  let expected_url = "https://github.com/CachedNerds/TestProject.git";
+  let actual_url = "https://github.com/CachedNerds/TestProject.git";
```
Comment (DannyPeck) [inline hatch/src/deps/tests/dependency.rs]: Shouldn't this be the method call on dep?

**Review by DannyPeck — CHANGES_REQUESTED** [review]

File: hatch/src/deps/tests/dependency.rs (hunk 4 above)
Comment (technetos) [reply hatch/src/deps/tests/dependency.rs]: Yup! I fixed it in #88 but ill fix it here too.

**Comment by technetos** [comment]
I have made the requested changes.

**Pull request closed without merging**